### PR TITLE
Update DWRF proto to support encrypted columns

### PR DESCRIPTION
**Pull request by rschlussel**

### CODE REVIEW

File: src/main/protobuf/dwrf_proto.proto
Comment (rschlussel) [inline src/main/protobuf/dwrf_proto.proto]: @zzhao0 can you clarify the status of this optimization (is it a part of the file format? what are the implications)

File: src/main/protobuf/dwrf_proto.proto
Comment (rschlussel) [inline src/main/protobuf/dwrf_proto.proto]: @zzhao0 can you clarify the status of this optimization (is it a part of the file format? what are the implications)

**Comment by rschlussel** [comment]
@zzhao0 @helfman 

@aweisberg do you know who has commit access to this repo, and also how it gets released, etc.?

Also, seems like the travis build doesn't work anymore either.

File: src/main/protobuf/dwrf_proto.proto
Comment (zzhao0) [reply src/main/protobuf/dwrf_proto.proto]: The optimization basically is same as that in open source ORC

We set keyMetadata once and have subsequent stripes (and the footer) reuse the key if key is not changed. This is part of the format. It's ok to set it for every single stripe and reader won't fail (if implemented correct to handle the runs)

File: src/main/protobuf/dwrf_proto.proto
Comment (zzhao0) [reply src/main/protobuf/dwrf_proto.proto]: if keyMetadata doesn't exist, that implied the entire file share the same key. Hence get key from the first stripe (key for first stripe must be set, since it's always start of the run)

if key metadata exists. That means it's a file with footer encrypted using it's own key (likely also means it's a concatenated file, with footer re-written). We then can use the key recorded in the footer as is

File: src/main/protobuf/dwrf_proto.proto
Comment (zzhao0) [inline src/main/protobuf/dwrf_proto.proto:107]: i would like to call out that offset follows the same optimization as keys when it only need to be set a the run start. If offset of one stream can be interpreted from offset of previous stream + length, then offset doesn't have to be set @rschlussel

**Comment by rschlussel** [comment]
@arhimondr @wenleix @mayankgarg1990

**Comment by rschlussel** [comment]
@zzhao0 I updated the code comments to reflect your explanations

**Review by arhimondr — APPROVED** [review]
Seems to be fine. Although I don't feel like I'm the right person to have a word on this. Feel free to merge it once @zzhao0 approves it.

**Comment by zzhao0** [comment]
I don’t have permission to approve. Looks good. Thanks!

**Comment by aweisberg** [comment]
@zzhao0 I added you to the contributor team and added the team to orc-protobuf

**Pull request merged**
